### ISSUE TITLE
Random example in F42

### ISSUE THREAD

**Issue by thecoshman** [isocpp/CppCoreGuidelines#332]
In [F42](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#-f42-return-a-t-to-indicate-a-position-only) there is an example of return references/pointers to varaibles about to go out of scope.

Whilst this is something to avoid, I don't see what it has to do with the rule at hand

**Comment by AndrewPardoe** [comment]
The rule's title is "Return a  T*  to indicate a position (only)". Maybe it would be clearer to title the rule as "Return a  T*  to indicate a position (only), never to transfer ownership" ?

**Comment by BjarneStroustrup** [comment]
There are two things a T* shouldn't do:
     (1) transfer ownership
     (2) be treated as an array

That doesn't fit in the title

**Comment by thecoshman** [comment]
Well, this seems to be part of a larger 'problem' with the rules about how to handle parameters and returns, and use of `owner` and `not_null`. There are some specific rules focusing on just one detail, such as something like "use `const T&` for in parameters too large to just pass by value", and then a rule that feels it needs to also explain how to handle the different ways of doing parameters.

Basically, it's failing at DRY.

**Comment by hsutter** [comment]
Hmm. I do think the rule statement is too limiting, perhaps because the example is mainly thinking of the specific case of an algorithm that returns `T*` as an iterator.

More generally, let's say an object A wants to return an object B that the caller should read or use and maintain B's identity (not copy). B might be an internal object that A owns, or it might be a distinct external object that A does not own and is observing. If null is an option, isn't returning `*` the natural thing? (BTW, it was hard to write the use case description without immediately saying `wants to return a pointer to...`)

**Comment by BjarneStroustrup** [comment]
I added a minor clarification and cross reference
